php/commands/drush.php

```diff
@@ -27,58 +28,77 @@ class Drush_Command extends CommandWithSSH {
-      $server['host'] = TERMINUS_HOST;
-    }
-
-    # Sanitize assoc args so we don't try to pass our own flags.
-    # TODO: DRY this out?
-    unset($assoc_args['site']);
-    if (isset($assoc_args['env'])) {
-      unset($assoc_args['env']);
-    }
-
-    # Create user-friendly output
     $command = implode( $args, ' ' );
-    $flags = '';
-    foreach ( $assoc_args as $k => $v ) {
-      if (isset($v) && (string) $v != '') {
-        $flags .= "--$k=$v ";
+    if (in_array($command, $this->special_commands)) {
+      $command_function     = str_replace('-', '_', $command);
+      $assoc_args['server'] = $server;
+      $result = $this->$command_function($assoc_args);
+    } else {
+      # Sanitize assoc args so we don't try to pass our own flags.
+      # TODO: DRY this out?
+      if (isset($assoc_args['site'])) {
+        unset($assoc_args['site']);
       }
-      else {
-        $flags .= "--$k ";
+      if (isset($assoc_args['env'])) {
+        unset($assoc_args['env']);
+      }
+
+      # Create user-friendly output
+      $flags = '';
+      foreach ( $assoc_args as $k => $v ) {
+        if (isset($v) && (string) $v != '') {
+          $flags .= "--$k=$v ";
+        }
+        else {
+          $flags .= "--$k ";
+        }
+      }
+      if (in_array(Terminus::getConfig('format'), array('bash', 'json', 'silent'))) {
+        $assoc_args['pipe'] = 1;
+      }
+      $this->log()->info(
+        "Running drush {cmd} {flags} on {site}-{env}",
+        array(
+          'cmd' => $command,
+          'flags' => $flags,
+          'site' => $site->get('name'),
+          'env' => $environment
+        )
+      );
+      $result = $this->sendCommand($server, 'drush', $args, $assoc_args);
+      if (Terminus::getConfig('format') != 'normal') {
+        $this->output()->outputRecordList($result);
       }
     }
-    if (in_array(\Terminus::getConfig('format'), array('bash', 'json', 'silent'))) {
-      $assoc_args['pipe'] = 1;
-    }
-    $this->log()->info(
-      "Running drush {cmd} {flags} on {site}-{env}",
-      array(
-        'cmd' => $command,
-        'flags' => $flags,
-        'site' => $site->get('name'),
-        'env' => $environment
-      )
-    );
-    $result = $this->sendCommand($server, 'drush', $args, $assoc_args);
-    if (Terminus::getConfig('format') != 'normal') {
-      $this->output()->outputRecordList($result);
-    }
+  }
+
+  /**
+   * Runs sql-connect via local drush installation
+   *
+   * @param [array]  $assoc_args Elements as follows:
+   *        [Site]   site        Site object for site to connect to
+   *        [string] environment Environment name
+   * @return [string] $result MySQL connection string
+   */
+  private function sql_connect($assoc_args) {
+    Terminus::launchSelf('sites', array('aliases'));
+    $site_name = $assoc_args['site'];
+    $env_name  = $assoc_args['env'];
+    ob_start();
+    passthru("drush @pantheon.$site_name.$env_name sql-connect");
```

Comment:
Come to think of it, I'm not sure why we don't run every Drush command this way. It's a lot less effort.